test/omeinsumcontractionordres.jl

```diff
@@ -0,0 +1,151 @@
+@testset "@tensor dependency check" begin
+    A = rand(2, 2)
+    B = rand(2, 2)
+    C = rand(2, 2)
+    @test_throws ArgumentError begin
+        ex = :(@tensor opt = (i => 2, j => 2, k => 2) opt_algorithm = GreedyMethod S[] := A[i,
+                                                                                            j] *
+                                                                                          B[j,
+                                                                                            k] *
+                                                                                          C[i,
+                                                                                            k])
+        macroexpand(Main, ex)
+    end
+end
+
+using OMEinsumContractionOrders
+
+# open the debug mode to check the optimization algorithms used
+ENV["JULIA_DEBUG"] = "TensorOperationsOMEinsumContractionOrdersExt"
```

Comment:
This does set this globally though, so if someone had this set to something else you would overwrite that here.
Perhaps you can make use of a `with_logger` clause to temporarily change the logger to enable debug messages?

Reply:
I tried the `@test_logs` or `with_logger`, it seems that it does not work correctly with the macro `@tensor`, may because of the precompiling.
I removed the global setting, and used `@test_logs` to test the `ncon` functions.